remotes.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+	"net/smtp"
+	"net/url"
+)
+
+type Remote struct {
+	SkipVerify bool
+	Auth       smtp.Auth
+	Scheme     string
+	Hostname   string
+	Port       string
+	Addr       string
+	Sender     string
+}
+
+// ParseRemote creates a remote from a given url in the following format:
+//
+// smtp://[user[:password]@][netloc][:port][/remote_sender][?param1=value1&...]
+// smtps://[user[:password]@][netloc][:port][/remote_sender][?param1=value1&...]
+//
+// Supported Params:
+// - skipVerify: can be "true" or empty to prevent ssl verification of remote server's certificate.
+// - auth: can be "login" to trigger "LOGIN" auth instead of "PLAIN" auth
+//
+func ParseRemote(remoteURL string) (*Remote, error) {
+	u, err := url.Parse(remoteURL)
+	if err != nil {
+		return nil, err
+	}
+
+	hostname, port := u.Hostname(), u.Port()
+
```

Comment:
There could be more cases ...

if u.Scheme == "" {
default smtp://
}

if port == "" {
smtp:// => default 25
smtps:// => default 465
starttls:// => default 587
}